### PR TITLE
Replace TxInput/boolean fields in ParsingModel with BlindVoteOutputState enum

### PR DESCRIPTION
**Pull request by chirhonul**

### CODE REVIEW

File: src/main/java/bisq/core/dao/node/parser/ParsingModel.java [hunk 1]
```diff
@@ -68,12 +68,17 @@
     @Nullable
     private OpReturnType opReturnTypeCandidate;
 
+    /**
+     * The different possible states for an output used in a blind vote.
+     */
+    enum BlindVoteOutputState {
```
Comment (ManfredKarrer) [inline src/main/java/bisq/core/dao/node/parser/ParsingModel.java]: I prefer to have enums or static inner classes at the top of the class

File: src/main/java/bisq/core/dao/node/parser/TxInputParser.java [hunk 2]
```diff
@@ -62,15 +62,13 @@ void process(TxInput txInput, int blockHeight, String txId, int inputIndex, Pars
                         case ISSUANCE_CANDIDATE_OUTPUT:
                             break;
                         case BLIND_VOTE_LOCK_STAKE_OUTPUT:
-                            if (parsingModel.getInputFromBlindVoteStakeOutput() == null) {
-                                parsingModel.setInputFromBlindVoteStakeOutput(txInput);
-                                // At the end of the parsing the OpReturnVoteRevealParser will verify that
-                                // this property is true, otherwise the tx will be considered invalid.
-                                parsingModel.setValidInputFromBlindVoteStakeOutput(true);
+                            if (parsingModel.getBlindVoteOutputState() == ParsingModel.BlindVoteOutputState.UNKNOWN) {
+                                // We found the input that's used as output for a blind vote. Only one such output is valid.
```
Comment (ManfredKarrer) [inline src/main/java/bisq/core/dao/node/parser/TxInputParser.java]: Maybe better: We found the input that's used as output of a blind vote. Only one such output is valid.

**Review by ManfredKarrer — CHANGES_REQUESTED** [review]

File: src/main/java/bisq/core/dao/node/parser/ParsingModel.java (hunk 1 above)
Comment (chirhonul) [reply src/main/java/bisq/core/dao/node/parser/ParsingModel.java]: Done, thanks.

File: src/main/java/bisq/core/dao/node/parser/TxInputParser.java (hunk 2 above)
Comment (chirhonul) [reply src/main/java/bisq/core/dao/node/parser/TxInputParser.java]: Reworded the comment.

**Comment by chirhonul** [comment]
Thanks, PTAL.

**Review by ManfredKarrer — APPROVED** [review]
utACK

**Pull request merged**